### PR TITLE
FFWEB-1928: Added information about `default-query` requirements and implications

### PR DESCRIPTION
**Pull request by a-laurowski**

### CODE REVIEW

File: markdown/3.x/en/api/ff-communication.api.md [hunk 1]
```diff
@@ -7,9 +7,9 @@ ___
 | **version**&nbsp;(String)&nbsp;(default: 7.2) | Your FACT-Finder version. Only major and minor version like "7.2" |
 | **api**&nbsp;(String)&nbsp;(default: `latest`) | Only relevant when `version="ng"`. The version of your FACT-Finder NG REST API. Possible values are `v2`, `v3` and `latest`. It is recommended to use an explicit version instead of `latest`. This avoids unintentionally breaking your application by updating your Web Components reference when a new API version becomes available. |
 | **channel**&nbsp;(String)&nbsp;(default: empty) | Your channel name. Has to be the same as the channel name configured in the FACT-Finder backend. |
-| **search-immediate**&nbsp;(Boolean) | If this property is present, FACT-Finder Web Components will start searching as soon as they are loaded. |
+| **search-immediate**&nbsp;(Boolean) | If this property is present, FACT-Finder Web Components will start searching as soon as they are loaded. For this to work, a `default-query` needs to be set with correct value (see `default-query` description) or should not be set at all. |
```
Comment (omikron-fp) [inline markdown/3.x/en/api/ff-communication.api.md:10]: Placing the reference to `default-query` at the end might allow a smoother reading flow.
```suggestion
| **search-immediate**&nbsp;(Boolean) | If this property is present, FACT-Finder Web Components will start searching as soon as they are loaded. For this to work, a `default-query` needs to be set with a correct value or to be not set at all (see `default-query` description). |
```

File: markdown/3.x/en/api/ff-communication.api.md [hunk 2]
```diff
@@ -7,9 +7,9 @@ ___
...
 | **use-url-parameters**&nbsp;(String) **Options**: &nbsp;"true", &nbsp;"false" (default: "true") | If set to false, FACT-Finder parameters are not pushed to the URL unless explicitly whitelisted. Note that this attribute has no effect when `only-search-params` is set. |
-| **default-query**&nbsp; (String) (default: '*') | Whenever a search is performed without a search query, this value will be used as default search term. |
+| **default-query**&nbsp; (String) (default: '*') | Whenever a search is performed without a search query, this value will be used as default search term. This attribute will not accept empty or whitespace-only string as its value. |
```
Comment (omikron-fp) [inline markdown/3.x/en/api/ff-communication.api.md:12]:
```suggestion
| **default-query**&nbsp; (String) (default: '*') | Whenever a search is performed without a search query, this value will be used as default search term. This attribute will not accept empty or whitespace-only strings as its value. |
```

**Review by omikron-fp — CHANGES_REQUESTED** [review]
Remarks apply to `4.x`, too.

File: markdown/3.x/en/api/ff-communication.api.md (hunk 2 above)
Comment (a-laurowski) [reply markdown/3.x/en/api/ff-communication.api.md]: > Remarks apply to `4.x`, too.
applied

**Review by omikron-fp — APPROVED** [review]

**Pull request merged**